File: AdminArea/AdminArea.Web/Modules/ServerTypes/onlinestore/categoryForm.ts
import { StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface categoryForm {
    Name: StringEditor;
    Photo: StringEditor;
    Description: StringEditor;
}

export class categoryForm extends PrefixedContext {
    static formKey = 'onlinestore.category';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!categoryForm.init)  {
            categoryForm.init = true;

            var w0 = StringEditor;

            initFormType(categoryForm, [
                'Name', w0,
                'Photo', w0,
                'Description', w0
            ]);
        }
    }
}

File: AdminArea/AdminArea.Web/Modules/onlinestore/category/categoryForm.cs
using Serenity.ComponentModel;

namespace AdminArea.onlinestore.Forms;

[FormScript("onlinestore.category")]
[BasedOnRow(typeof(categoryRow), CheckNames = true)]
public class categoryForm
{
    public string Name { get; set; }
    public string Photo { get; set; }
    public string Description { get; set; }
}